odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/reactive/ReactiveDatasetVersionRepositoryImpl.java

```diff
@@ -308,19 +308,17 @@ private DatasetVersionFields mapToDatasetVersionFields(final Record record,
         return new DatasetVersionFields(version, fields);
     }
 
-    private List<LabelDto> extractLabels(final Record record) {
-        final Set<LabelPojo> labels = jooqRecordHelper.extractAggRelation(record, LABELS, LabelPojo.class);
+    private List<TagDto> extractTags(final Record record) {
+        final Set<TagPojo> tags = jooqRecordHelper.extractAggRelation(record, TAGS, TagPojo.class);
 
-        final Map<Long, LabelToDatasetFieldPojo> relations = jooqRecordHelper
-            .extractAggRelation(record, LABEL_RELATIONS, LabelToDatasetFieldPojo.class)
+        final Map<Long, TagToDatasetFieldPojo> relations = jooqRecordHelper
+            .extractAggRelation(record, TAG_RELATIONS, TagToDatasetFieldPojo.class)
             .stream()
-            .collect(Collectors.toMap(LabelToDatasetFieldPojo::getLabelId, identity()));
+            .collect(Collectors.toMap(TagToDatasetFieldPojo::getTagId, identity()));
 
-        return labels.stream()
-            .map(labelPojo -> new LabelDto(
-                labelPojo,
-                !LabelOrigin.INTERNAL.equals(LabelOrigin.valueOf(relations.get(labelPojo.getId()).getOrigin()))
-            ))
+        return tags.stream()
+            .map(tagPojo -> new TagDto(tagPojo, null,
+                TagOrigin.EXTERNAL.name().equals(relations.get(tagPojo.getId()).getOrigin())))
```

Comment:
please make it !TagOrigin.INTERNAL.name().equals(...), because we have EXTERNAL and EXTERNAL_STATISTICS